tests/test_hlsstream.py

```diff
@@ -0,0 +1,459 @@
+        except Exception as e:
+            print(f"  [FAIL] First call raised exception: {e}")
+            sys.exit(1)
+
+        # Test 2: Second call should advance the time window
+        print("  Testing second get_next_clip call...")
+        current_clip_end_time = clip_end1 + timedelta(0, polling_interval)
+
+        print(f"    Second timestamp: {current_clip_end_time}")
+        try:
+            wav_path2, clip_start2, clip_end2 = stream.get_next_clip(
+                current_clip_end_time
+            )
+            if wav_path2 is None:
+                print(
+                    "  [INFO] Second call returned None "
+                    "(stream unavailable or insufficient data)"
+                )
+                second_call_success = False
+            else:
+                print("  [PASS] Second call retrieved clip")
+                print(f"    Start: {clip_start2}, End: {clip_end2}")
+                second_call_success = True
+        except Exception as e:
+            print(f"  [FAIL] Second call raised exception: {e}")
+            sys.exit(1)
+
+        # Test 3: Verify clips don't overlap (if both succeeded)
+        if first_call_success and second_call_success:
+            print("  Verifying clip times don't overlap...")
+            # Parse the ISO format timestamps if they're strings
+            if isinstance(clip_start2, str):
+                start2_dt = datetime.fromisoformat(clip_start2.rstrip("Z"))
+            else:
+                start2_dt = clip_start2
+
+            # Check if second clip starts after or at the same time as first
+            if isinstance(clip_end1, datetime) and isinstance(
+                start2_dt, datetime
+            ):
+                if start2_dt >= clip_end1:
+                    print(
+                        "  [PASS] Clips don't overlap - "
+                        "time window advanced properly"
+                    )
+                else:
+                    print(
+                        f"  [FAIL] Clips overlap: "
+                        f"clip1 ends at {clip_end1}, "
+                        f"clip2 starts at {start2_dt}"
+                    )
+                    sys.exit(1)
+            else:
+                print("  [INFO] Could not verify overlap (timestamp formats)")
+        else:
+            print(
+                "  [INFO] Skipping overlap verification "
+                "(one or both calls did not retrieve clips)"
+            )
+
+        print(
+            "  [PASS] Sequential get_next_clip calls "
+            "completed without crashes"
+        )
+
+    finally:
+        # Clean up test directory
+        if os.path.exists(wav_dir):
+            shutil.rmtree(wav_dir)
+
+    print("[PASS] Sequential clip retrieval tests completed")
+
+
+def main():
+    """Run all tests."""
+    print("=" * 60)
+    print("Running HLSStream Tests")
+    print("=" * 60)
+
+    # Default stream base for tests
+    default_stream_base = (
```

Comment:
these can be defined as fixtures so that all tests use them while running pytest.